core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2010,11 +2010,21 @@ public void deleteInternal(RpcContext rpcContext, LockedInodePath inodePath,
       }
 
       if (!failedUris.isEmpty()) {
-        Collection<String> messages = failedUris.stream()
-            .map(pair -> String.format("%s (%s)", pair.getFirst(), pair.getSecond()))
-            .collect(Collectors.toList());
-        throw new FailedPreconditionException(
-            ExceptionMessage.DELETE_FAILED_UFS.getMessage(StringUtils.join(messages, ", ")));
+        StringBuilder errorReport = new StringBuilder(
+            ExceptionMessage.DELETE_FAILED_UFS.getMessage(failedUris.size() + " paths: "));
+        boolean trim = failedUris.size() > 20;
```

Comment:
If 20 is not enough, we can change it to 100. For numbers more than 100, I don't think people will read those additional information carefully.

Reply:
Could you move this logic to a private method? `buildDeleteFailureMessage()`?

Reply:
Done